gravitee-am-reporter/gravitee-am-reporter-kafka/src/test/java/io/gravitee/am/reporter/kafka/audit/KafkaAuditReporterWithSchemaRegistryITests.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+import org.testcontainers.containers.KafkaContainer;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = JUnitConfiguration.class, loader = AnnotationConfigContextLoader.class)
+public class KafkaAuditReporterWithSchemaRegistryITests {
+
+    private KafkaAuditReporter kafkaAuditReporter;
+
+    @Autowired
+    protected ApplicationContext context;
+
+    @Autowired
+    protected KafkaContainer kafkaContainer;
+
+    @Autowired
+    protected SchemaRegistryContainer schemaRegistryContainer;
+
+    @Autowired
+    @Qualifier("configWithSchemaRegistry")
+    protected KafkaReporterConfiguration configuration;
+
+    @BeforeEach
+    void init() throws Exception {
+        this.kafkaAuditReporter = new KafkaAuditReporter(configuration);
+
+        context.getAutowireCapableBeanFactory().autowireBean(kafkaAuditReporter);
+
+        this.kafkaAuditReporter.afterPropertiesSet();
+        this.kafkaAuditReporter.start();
+    }
+
+    @Test
+    void shouldWriteEventAsJson_When_EventIsAudit_WithSchemaRegistry() throws Exception {
+        this.kafkaAuditReporter.report(AuditValueFactory.createAudit());
+
+        Map<String, Object> config = new HashMap<>();
+        config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, this.kafkaContainer.getBootstrapServers());
+        config.put(ConsumerConfig.GROUP_ID_CONFIG, "junit-" + UUID.randomUUID());
+        config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        String schemaRegistryUrl = "http://" + schemaRegistryContainer.getHost() + ":" + schemaRegistryContainer.getFirstMappedPort();
+        config.put("schema.registry.url", schemaRegistryUrl);
+
+        Deserializer<String> keyDeserializer = new StringDeserializer();
+        KafkaJsonSchemaDeserializer<AuditMessageValueDto> valueDeserializer = getDeserializer();
+        valueDeserializer.configure(config, false);
+
+        KafkaConsumer<String, AuditMessageValueDto> kafkaConsumer = new KafkaConsumer<>(config, keyDeserializer, valueDeserializer);
+        kafkaConsumer.subscribe(Collections.singletonList(this.configuration.getTopic()));
+
+        ConsumerRecords<String, AuditMessageValueDto> poll = kafkaConsumer.poll(Duration.of(10, ChronoUnit.SECONDS));
+        assertNotNull(poll);
+        assertEquals(1, poll.count());
+
+        ConsumerRecord<String, AuditMessageValueDto> record = poll.iterator().next();
```

Comment:
question: where do we check that the schema id is present into the record ?

Reply:
There is specified `schema.registry.url` in consumer properties. If this wouldn't be provided, than consumer don't know how to validate te message and throw an error. When we consume this message as a string, we can see at the beginning few characters that correspond to schema Id.